cdi/embedded/src/main/java/org/infinispan/cdi/embedded/DelegatingDistributedTaskLifecycle.java

```diff
@@ -44,13 +45,19 @@ public DelegatingDistributedTaskLifecycle() {
 
       @Override
       public <T, K, V> void onPreExecute(Callable<T> task, Cache<K, V> inputDataCache) {
-         BeanManager bm = CDIHelper.getBeanManager();
-         if (bm == null)
-            return;
-         ContextInputCache.set(inputDataCache);
-         preInject(bm, task);
-         if (task instanceof RunnableAdapter) {
-            preInject(bm, ((RunnableAdapter) task).getTask());
+         try {
+            BeanManager bm = CDIHelper.getBeanManager();
+            if (bm == null)
+               return;
+            ContextInputCache.set(inputDataCache);
+            preInject(bm, task);
+            if (task instanceof RunnableAdapter) {
+               preInject(bm, ((RunnableAdapter) task).getTask());
+            }
+         } catch (Exception e) {
+            throw new IllegalStateException("Injcting input caches failed. This may result in NullPointerExceptions in client code", e);
```

Comment:
Typo: Injcting

Also, I think continuing to run the task after throwing the `IllegalStateException` would be a bug, as the NPE in the user code would hide the ISE.